ts/ui/lazy/LazyHandler.ts

```diff
@@ -346,9 +355,80 @@ B extends MathDocumentConstructor<HTMLDocument<N, T, D>>>(
       this.lazyObserver = new IntersectionObserver(this.lazyObserve.bind(this));
       this.lazyList = new LazyList<N, T, D>();
       const callback = this.lazyHandleSet.bind(this);
-      this.lazyProcessSet = (typeof window !== 'undefined' && window.requestIdleCallback ?
-                          () => window.requestIdleCallback(callback) :
-                          () => setTimeout(callback, 10));
+      this.lazyProcessSet = (window && window.requestIdleCallback ?
+                             () => window.requestIdleCallback(callback) :
+                             () => setTimeout(callback, 10));
+      //
+      //  Install print listeners to typeset the rest of the document before printing
+      //
+      if (window) {
+        let done = false;
+        const handler = () => {
+          !done && this.lazyTypesetAll();
+          done = true;
+        };
+        window.matchMedia('print').addListener(handler);  // for Safari
```

Comment:
That's curious. MDN says Safari knows `beforeprint`. But I am sure you've tested it.

Reply:
It was added in Safari v13, which is not all that long ago.  I wrote the code last year, and may have been running 12 at that point.  I can remove the support if you want, but it may help with other browsers that I haven't tested.  Removing it would mean that the previous lines could be simplified (since we wouldn't have to check for duplicate calls), so perhaps that is worth it.

Reply:
I tested that using only `before print` does work in the current Safari.  I suppose the real question then is how much backward-compatibility do we need?  There may be WebKit-based browsers (or tools lie WebView) that are not as up-to-date as Safari itself.

Reply:
Let's keep it then.